src/modules/avformat/producer_avformat.c

```diff
@@ -1907,6 +1907,7 @@ static int producer_get_image( mlt_frame frame, uint8_t **buffer, mlt_image_form
 								if( transfer_data_result < 0 ) 
 								{
 									mlt_log_error( MLT_PRODUCER_SERVICE( producer ), "av_hwframe_transfer_data() failed %d\n", transfer_data_result );
+									mlt_service_unlock(MLT_PRODUCER_SERVICE(producer));
```

Comment:
Same as above.

Reply:
Yes, thanks

Reply:
> Wait, my branch of this says `return -1` after this line instead of `goto exit_get_image`

Forgot to mention I changed this as well. `goto` is probably the proper way to go about it here, since simply returning would leave a bunch of stuff in an inconsistent state.